### ISSUE TITLE
SC1102 false positive: Using variable as arithmetic operator

### ISSUE THREAD

**Issue by aureliojargas** [koalaman/shellcheck#2000]
#### For bugs
- Rule Id: SC1102
- My shellcheck version: 0.7.1
- [X] The rule's wiki page does not already cover this
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
op='+'
echo "$((1 $op 2))"
```

#### Here's what shellcheck currently says:

```
Line 3:
echo "$((1 $op 2))"
     ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
      ^-- SC1102: Shells disambiguate $(( differently or not at all. For $(command substition), add space after $( . For $((arithmetics)), fix parsing errors.
```

#### Here's what I wanted or expected to see:

No warnings.

Using an arithmetic operator inside a variable is fine in busybox's `sh` and in bash 5:

```console
$ docker run --rm -v "$PWD:/mnt" alpine sh /mnt/op.sh 
3
$ bash op.sh 
3
$
```

**Comment by aureliojargas** [comment]
From my side, I refactored my code to not use the arithmetic operator as a variable. Since this is not an issue for me anymore, I'm closing this one. Feel free to reopen if you think it's relevant anyway.
